domain/src/test/java/org/oppia/android/domain/topic/TopicListControllerTest.kt

```diff
@@ -462,6 +463,58 @@ class TopicListControllerTest {
     )
   }
 
+  @Test
+  fun testGetStoryList_markRecentlyPlayedforFirstTestTopic_promotedListIsCorrect() {
+    storyProgressController.recordRecentlyPlayedChapter(
+      profileId0,
+      TEST_TOPIC_ID_0,
+      TEST_STORY_ID_0,
+      TEST_EXPLORATION_ID_2,
+      getCurrentTimestamp()
+    )
+    testCoroutineDispatchers.runCurrent()
+
+    val promotedActivityList = retrievePromotedActivityList()
+    assertThat(promotedActivityList.promotedStoryList.recentlyPlayedStoryCount)
+      .isEqualTo(1)
+    verifyOngoingStoryAsFirstTopicStory0Exploration0(
+      promotedActivityList.promotedStoryList.recentlyPlayedStoryList[0]
+    )
+    assertThat(promotedActivityList.promotedStoryList.suggestedStoryCount)
+      .isEqualTo(1)
+    verifyPromotedStoryAsFractionStory0Exploration0(
+      promotedActivityList.promotedStoryList.suggestedStoryList[0]
+    )
+  }
+
+  @Test
+  fun testGetStoryList_markOneStoryDoneForFirstTestTopic_suggestedListIsCorrect() {
+    storyProgressController.recordCompletedChapter(
+      profileId0,
+      TEST_TOPIC_ID_0,
+      TEST_STORY_ID_0,
+      TEST_EXPLORATION_ID_2,
+      getCurrentTimestamp()
+    )
+    testCoroutineDispatchers.runCurrent()
+
+    storyProgressController.recordCompletedChapter(
+      profileId0,
+      TEST_TOPIC_ID_0,
+      TEST_STORY_ID_0,
+      TEST_EXPLORATION_ID_5,
+      getCurrentTimestamp()
+    )
+    testCoroutineDispatchers.runCurrent()
+
+    val promotedActivityList = retrievePromotedActivityList()
+    assertThat(promotedActivityList.promotedStoryList.suggestedStoryCount)
+      .isEqualTo(1)
+    verifyPromotedStoryAsFractionStory0Exploration0(
```

Comment:
This seems backwards. Topic 0 depends on fractions, that means when fractions is completed topic 0 should be recommended not the other way around. Is the matching logic correct?

Reply:
I have added it other way around. Logic seems correct. I will change it.

Reply:
See my comment below--I don't think this is right (or at least it seems inconsistent since the direction is one way for some of the topics, but reversed for the test topics). Closing this comment thread in favor of the other.

Reply:
Still unresolved. Recommended topics for this scenario (finishing test topic 1) should be just ratios.